File: 7-ReverseInteger/7-ReverseInteger.py
# Last updated: 6/22/2026, 12:55:20 PM
class Solution:
    def reverse(self, x: int) -> int:
        if x < 0 :
            rev = -1*int(str(x)[-1:0:-1])
            return rev if -1*(2**31)<rev<2**31 else 0
        rev = int(str(x)[::-1])
        return rev if -1*(2**31)<rev<2**31 else 0
        

File: 7-ReverseInteger/7-ReverseInteger.js
// Last updated: 6/22/2026, 12:55:27 PM
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
    if(x<Math.pow(2,31) && x>= Math.pow(2,31)) return 0
    let num = x.toString().split('').toReversed().map(e => parseInt(e))
    if (x < 0) {
        num.pop();
        num = -1 * parseInt(num.join(''))
        return num >= -1 * Math.pow(2,31) ? num : 0;
    } else {
        num = parseInt(num.join(''))
        return num < Math.pow(2,31) ? num : 0;
    }
};